File: app/src/main/java/com/tosh/poolandroid/ViewModel/PhoneViewModel.java
package com.tosh.poolandroid.ViewModel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;

import com.tosh.poolandroid.Remote.AuthRetrofitClient;
import com.tosh.poolandroid.Remote.NodeAuthService;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;


public class PhoneViewModel extends AndroidViewModel {

    private static MutableLiveData<String> phoneStatus = new MutableLiveData<>();

    private NodeAuthService api;
    private CompositeDisposable compositeDisposable = new CompositeDisposable();


    public PhoneViewModel(@NonNull Application application) {
        super(application);
        api = AuthRetrofitClient.getInstance().create(NodeAuthService.class);
    }

    public void getPhone(String name, String phone, String email) {
        compositeDisposable.add(api.sendSms(name,phone, email)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Consumer<String>() {
                    @Override
                    public void accept(String res) throws Exception {

                        if (res.equals("phone")){
                            phoneStatus.postValue("success");
                        }else {
                            phoneStatus.postValue(res);
                        }
                    }
                })
        );

    }

    public MutableLiveData<String> getPhoneStatus(){
        return phoneStatus;
    }

    @Override
    protected void finalize() throws Throwable {
        super.finalize();
        compositeDisposable.clear();
    }
}

File: app/src/main/java/com/tosh/poolandroid/viewmodel/PhoneViewModel.kt
package com.tosh.poolandroid.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.tosh.poolandroid.model.RegisterResponse
import com.tosh.poolandroid.Remote.RetrofitClient
import com.tosh.poolloginrebuild.database.UserEntity
import com.tosh.poolloginrebuild.repository.UserRepository
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class PhoneViewModel(application: Application) :AndroidViewModel(application) {
    private val repository: UserRepository = UserRepository(application)

    fun addUserPhone(name: String, email: String, phone:String): LiveData<String> {
        val registerResponse = MutableLiveData<String>()

        RetrofitClient.makeRetrofitApi().addUserPhone(name, phone, email)
            .enqueue(object : Callback<RegisterResponse> {
                override fun onFailure(call: Call<RegisterResponse>, t: Throwable) {
                    registerResponse.value = t.message
                }

                override fun onResponse(call: Call<RegisterResponse>, response: Response<RegisterResponse>) {
                    if (response.isSuccessful) {
                        registerResponse.value = response.body()?.message.toString()

                        if (response.body()?.message == "successful") {

                                val newUser = UserEntity(
                                    name = response.body()!!.user.name,
                                    email = response.body()!!.user.email,
                                    phone = response.body()!!.user.phone
                                )

                                insert(newUser)
                        }
                    } else {
                        registerResponse.value = response.body()?.message.toString()

                    }
                }

            })
        return registerResponse
    }

    fun insert(userEntity: UserEntity) {
        repository.insert(userEntity)
    }
}